tools/linter/fetcher.rb

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'api/compiler'
+
+# Takes in a DiscoveryResource + Api::Resource
+# Loops through all properties of the DiscoveryResource (at any depth)
+# Passes the DiscoveryProperties and their corresponding Api Properties
+# to a test block.
+class PropertyFetcher
+  def initialize(discovery_res, api_res)
+    @discovery_res = discovery_res
+    @api_res = api_res
+  end
+
+  def run(&block)
```

Comment:
I'm having trouble seeing why this is a class and not just a function?

`compare_schema_and_docs_with_test_block(discovery_res, api_res, &block)`

or something along those lines.

Reply:
Covering both function-or-class comments. What about using a module, or whatever the equivalent of a static class is to namespace the functions? That way we don't need class semantics for what's really just a function call.